Doc/c-api/hash.rst

```diff
@@ -89,6 +89,24 @@ See also the :c:member:`PyTypeObject.tp_hash` member and :ref:`numeric-hash`.
 
    .. versionadded:: 3.13
 
+
+.. c:function:: Py_hash_t Py_HashBytes(const void *ptr, Py_ssize_t len)
+
+   Compute and return the hash value of a buffer of *len* bytes
+   starting at address *ptr*. The hash is guaranteed to match that of
+   :class:`bytes`, :class:`memoryview`, and other built-in objects
+   that implement the :ref:`buffer protocol <bufferobjects>`.
+
+   Use this function to implement hashing for immutable objects whose
+   ``tp_richcompare`` function compares to another object's buffer.
+
+   *len* must be greater than or to equal to ``0``.
+
+   The function cannot fail: it cannot return ``-1``.
```

Comment:
We normally use this wording in the C API docs.

```suggestion
   This function always succeeds.
```

Reply:
+1. We standardized on -1/NULL for *new* API, but we shouldnot suggest that all C API uses that.